src/ape_ethereum/ecosystem.py

```diff
@@ -559,36 +558,31 @@ def decode_receipt(self, data: dict) -> ReceiptAPI:
             "transactionHash",
             "transaction_hash",
         )
-        for choice in hash_key_choices:
-            if choice in data:
-                txn_hash = data[choice]
-                break
+        txn_hash = next((data[choice] for choice in hash_key_choices if choice in data), None)
+        if txn_hash and isinstance(txn_hash, bytes):
+            txn_hash = txn_hash.hex()
 
-        if txn_hash:
-            txn_hash = txn_hash.hex() if isinstance(txn_hash, bytes) else txn_hash
-
-        data_bytes = data.get("data", b"")
+        data_bytes = data.get("data")
         if data_bytes and isinstance(data_bytes, str):
             data["data"] = HexBytes(data_bytes)
-
         elif "input" in data and isinstance(data["input"], str):
             data["input"] = HexBytes(data["input"])
 
         block_number = data.get("block_number", data.get("blockNumber"))
         if block_number is None:
             raise ValueError("Missing block number.")
 
-        receipt_kwargs = dict(
```

Comment:
I kinda like this method better, is there some tradeoff vs. Doing it as a dict literal?

Reply:
Yes, it seems significant enough:

```
(ape08) ➜  ape git:(perf/pytest) ✗ python -m timeit "dict()"
5000000 loops, best of 5: 54.6 nsec per loop
(ape08) ➜  ape git:(perf/pytest) ✗ python -m timeit "{}"
10000000 loops, best of 5: 30.5 nsec per loop
```

Reply:
This is also kind of interesting:

```python
>>> from dis import dis
>>> dis("dict()")
  0           0 RESUME                   0

  1           2 PUSH_NULL
              4 LOAD_NAME                0 (dict)
              6 CALL                     0
             14 RETURN_VALUE
>>> dis("{}")
  0           0 RESUME                   0

  1           2 BUILD_MAP                0
              4 RETURN_VALUE
```

I using Python 3.12 btw, older versions I heard it was even worse.

Reply:
Wow